pkg/output/printer.go

```diff
@@ -43,27 +41,35 @@ const (
 )
 
 type PrintOptions struct {
-	Fields      []string
-	FieldsLong  []string
-	IgnoreFlags bool
-	Output      OutputOption
-	Pager       io.Writer
-	NoPager     bool
-	NoHeader    bool
-	Separator   string
+	// Fields is a list of fields to print
+	Fields []string
+	// FieldsLong is a list of additional fields to print with "--fields long" flag
+	FieldsLong []string
+	// ForceFields ignores user provided fields and uses print options instead. Useful when printing secondary data
+	ForceFields bool
```

Comment:
Is this used in anything other than secondary data? Maybe just call it what it is and have a flag in the formatting functions to mark data as secondary?

Reply:
so the Secondary Data flag would make 
- in Table view: ignore user provided `--fields` (because the flag is meant to be applied to the main data)
- in JSON view: don't print secondary data at all to make the main data JSON valid for example for history replay?

Reply:
still merging the current PR as the PR targets output writer and not exactly the formatting API

Reply:
I'd need to understand the code a bit better to provide a concrete suggestion here.